### ISSUE TITLE
Metrics tags for virtual cluster / broker

### ISSUE THREAD

**Issue by k-wall** [kroxylicious/kroxylicious#264]
**Is your feature request related to a problem? Please describe.**

#254 changes kroxy so that it is capable of expressing one or more virtual clusters and adds support for clusters formed of more than one broker.

The metrics emitted by kroxy should include tags that identify the virtual cluster (name/clusterid) so that the user of the metrics is able to differentiate, say the downstream message count from one virtual cluster from another.

Furthermore it should be possible to break the metrics down by broker.   As user I should be able to know the downstream message count for broker 0 from that of broker 1.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

Netty Channels can carry an attribute.  I'm wondering if that mechanism might be a suitable way to carry the

**Comment by k-wall** [comment]
@racheljpg @robobario @SamBarker would love to get input here to help shape this feature required.  Feel free to edit the issue.

**Comment by robobario** [comment]
I had a [hack around](https://github.com/kroxylicious/kroxylicious/commit/8b8a4cc3eecc4f925ee3e89506831cbf04830907) on top of your branch adding `VirtualCluster` @k-wall 

Rather than using Channel attributes to pass around tags or some other context I've been playing with adding in an idea of `ContextualProxyMetrics`. This is a class we instantiate for every `VirtualCluster` which can be used to obtain a micrometer MeterRegistry with additional tags configured into it. I also wired this into the `KrpcFilterContext` so filter authors could also have the ability to record metrics at these different granularities.

In my branch the `BrokerAddressFilter` is using it like:

```
context.metrics().getVirtualClusterTagged().counter("broker_address_applications").increment();
```
which is scraped from `/metrics` as:
```
broker_address_applications_total{virtual_cluster_name="demo",} 1.0
```

So I'd imagine also having a method like: `getVirtualClusterTargetClusterTagged()` which has tags for virtual cluster and target cluster identifiers.

This would allow us to evolve the tags added for each granularity without having to update the users of those registries.

A risk that I've seen before is that the underlying prometheus registry does not tolerate us registering the same metrics with different [sets of tags](https://github.com/kroxylicious/kroxylicious/pull/122#issuecomment-1413093573). It will (quietly) not expose some recorded data when scraped. So we must be careful to not register the same metric name with different sets of tags. Like:
```
context.metrics().getVirtualClusterTagged().counter("request_total").increment();
context.metrics().getVirtualClusterTargetClusterTagged().counter("request_total").increment();
Metrics.counter("request_total").increment()
```
Only one of the counts would be scrape-able, because they would all have different tag names.

Maybe we can configure something into the registry to detect this and make it complain but at the moment it's a quiet failure.

**Comment by k-wall** [comment]
Your ideas make sense to me.

**Comment by k-wall** [comment]
Seems to be a duplicate of #1109

**Comment by SamBarker** [comment]
I agree this is a dupe of https://github.com/kroxylicious/kroxylicious/issues/1109 and #1109 is clearer so closing in favour of that.
